✅ Pass lexing a single eberban space

Benjamin-Piper opened the pull request.

Benjamin-Piper committed:
✅ Pass lexing a single eberban space

mia-entropy commented on web/src/scripts/lexer/index.ts:
```diff
@@ -1,10 +1,33 @@
+}
+
+const is_repeat = (char: string, last_char: string): boolean => {
+    const both_are_identical = char === last_char;
```
This will not work if different allowed characters/codepoints for the same char is used. Example: "eEe"

Benjamin-Piper replied:
Fixed

Benjamin-Piper committed:
➕ Install confusables

When lexing single characters (except within foreign quotes) and checking
for repeats, we need to consider when there is:

- different case (a vs A)
- different composition  (á vs a+')
- look-alikes (Latin A vs Greek alpha)

To do this, we need something stronger than a `===` string check.

So we've chosen to use the https://github.com/gc/confusables package to
help with this.

Benjamin-Piper committed:
👔 Deem consecutive confusables to be repeats

Note: the set of confusable characters does not
include whitespace. I'm not sure why, but that's how it is.

We'll now be checking for look-alike characters and characters of
different cases. However, because we're not lexing words, no tests have
been written for them right now.

Benjamin-Piper committed:
♻️ Clarify that repeated characters are not empty
